src/cli/main.rs

```diff
@@ -341,12 +364,35 @@ fn format(opt: opt::Opt) -> Result<i32> {
                     let tx = tx.clone();
                     let opt = opt.clone();
 
+                    let should_skip_format = match &opt.stdin_filepath {
+                        Some(filepath) => {
+                            let ignore = get_ignore(
+                                filepath.parent().expect("cannot get parent directory"),
+                                opt.search_parent_directories,
+                            )
+                            .context("failed to parse ignore file");
```

Comment:
Sorry, the `?` was meant to be part of it
```suggestion
                            .context("failed to parse ignore file")?;
```

And then you won't need to match on the result

Not sure if this works though

Reply:
Yep, it works and simplifies things 🎉